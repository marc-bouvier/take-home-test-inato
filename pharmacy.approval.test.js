import { Drug, Pharmacy } from "./pharmacy";
import { verify } from "approvals/lib/Providers/Jest/JestApprovals";

describe("Pharmacy", () => {
  it("approve main case", () => {
    const drugs = [
      new Drug("test", 1, 2)];
    const trial = new Pharmacy(drugs);

    const log = [];
    for (let elapsedDays = 0; elapsedDays < 30; elapsedDays++) {
      log.push(JSON.stringify(trial.updateBenefitValue()));
    }

    /* eslint-disable no-console */
    verify(log.toString());
  });

});
