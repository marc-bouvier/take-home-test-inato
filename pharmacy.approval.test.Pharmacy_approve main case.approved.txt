[{"name":"test","expiresIn":0,"benefit":1}],[{"name":"test","expiresIn":-1,"benefit":0}],[{"name":"test","expiresIn":-2,"benefit":0}],[{"name":"test","expiresIn":-3,"benefit":0}],[{"name":"test","expiresIn":-4,"benefit":0}],[{"name":"test","expiresIn":-5,"benefit":0}],[{"name":"test","expiresIn":-6,"benefit":0}],[{"name":"test","expiresIn":-7,"benefit":0}],[{"name":"test","expiresIn":-8,"benefit":0}],[{"name":"test","expiresIn":-9,"benefit":0}],[{"name":"test","expiresIn":-10,"benefit":0}],[{"name":"test","expiresIn":-11,"benefit":0}],[{"name":"test","expiresIn":-12,"benefit":0}],[{"name":"test","expiresIn":-13,"benefit":0}],[{"name":"test","expiresIn":-14,"benefit":0}],[{"name":"test","expiresIn":-15,"benefit":0}],[{"name":"test","expiresIn":-16,"benefit":0}],[{"name":"test","expiresIn":-17,"benefit":0}],[{"name":"test","expiresIn":-18,"benefit":0}],[{"name":"test","expiresIn":-19,"benefit":0}],[{"name":"test","expiresIn":-20,"benefit":0}],[{"name":"test","expiresIn":-21,"benefit":0}],[{"name":"test","expiresIn":-22,"benefit":0}],[{"name":"test","expiresIn":-23,"benefit":0}],[{"name":"test","expiresIn":-24,"benefit":0}],[{"name":"test","expiresIn":-25,"benefit":0}],[{"name":"test","expiresIn":-26,"benefit":0}],[{"name":"test","expiresIn":-27,"benefit":0}],[{"name":"test","expiresIn":-28,"benefit":0}],[{"name":"test","expiresIn":-29,"benefit":0}]
